### ISSUE TITLE
Can this be ported to Scala 3 / Dotty

### ISSUE THREAD

**Issue by sirinath** [adamw/scala-macro-aop#2]
@adamw  Can this be made into a polished Dotty / Scala 3 implementation? Also perhaps make to more fine-grain so one can use any Scala / Dotty construed as an advice.

This can prove useful.

**Comment by adamw** [comment]
I don't think so, there are no macro annotations in Dotty, so instrumenting a whole class is probably impossible. We only have `inline` + quoting/splicing: https://dotty.epfl.ch/docs/reference/metaprogramming/macros.html

**Comment by sirinath** [comment]
What about plugins?

**Comment by adamw** [comment]
Good question - I don't know :)
